crates/core/src/factories/core_fetch/test/response_queue.rs

```diff
@@ -0,0 +1,228 @@
+        &self,
+        _peer: Url,
+        _space: SpaceId,
+        _data: bytes::Bytes,
+    ) -> kitsune2_api::BoxFut<'_, kitsune2_api::K2Result<()>> {
+        unimplemented!()
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn respond_to_multiple_requests() {
+    let builder = Arc::new(default_builder().with_default_config().unwrap());
+    let peer_store = builder.peer_store.create(builder.clone()).await.unwrap();
+    let op_store = Arc::new(Kitsune2MemoryOpStore::default());
+    let mock_transport = MockTransport::new();
+    let config = CoreFetchConfig::default();
+    let space_id = SpaceId::from(random_id());
+
+    let op_id_1 = random_op_id();
+    let op_id_2 = random_op_id();
+    let agent_id_1 = random_agent_id();
+    let agent_info_1 = AgentBuilder {
+        agent: Some(agent_id_1.clone()),
+        url: Some(Some(Url::from_str("wss://127.0.0.1:1").unwrap())),
+        ..Default::default()
+    }
+    .build();
+    let agent_url_1 = agent_info_1.url.clone().unwrap();
+    peer_store.insert(vec![agent_info_1.clone()]).await.unwrap();
+
+    let op_id_3 = random_op_id();
+    let op_id_4 = random_op_id();
+    let agent_id_2 = random_agent_id();
+    let agent_info_2 = AgentBuilder {
+        agent: Some(agent_id_2.clone()),
+        url: Some(Some(Url::from_str("wss://127.0.0.1:2").unwrap())),
+        ..Default::default()
+    }
+    .build();
+    let agent_url_2 = agent_info_2.url.clone().unwrap();
+    peer_store.insert(vec![agent_info_2.clone()]).await.unwrap();
+
+    // Insert ops to be read and sent into op store.
+    let op_1 = MetaOp {
+        op_id: op_id_1.clone(),
+        op_data: serde_json::to_vec(&Kitsune2MemoryOp::new(
+            op_id_1.clone(),
+            Timestamp::now(),
+            vec![1; 128],
+        ))
+        .unwrap(),
+    };
+    let op_2 = MetaOp {
+        op_id: op_id_2.clone(),
+        op_data: serde_json::to_vec(&Kitsune2MemoryOp::new(
+            op_id_2.clone(),
+            Timestamp::now(),
+            vec![2; 128],
+        ))
+        .unwrap(),
+    };
+    let op_3 = MetaOp {
+        op_id: op_id_3.clone(),
+        op_data: serde_json::to_vec(&Kitsune2MemoryOp::new(
+            op_id_3.clone(),
+            Timestamp::now(),
+            vec![3; 128],
+        ))
+        .unwrap(),
+    };
+    // Insert op 1, 2, 3 into op store. Op 4 will not be returned in the response.
+    let ops_to_store = vec![op_1.clone(), op_2.clone(), op_3.clone()];
+    op_store.process_incoming_ops(ops_to_store).await.unwrap();
+
+    let fetch = CoreFetch::new(
+        config.clone(),
+        space_id.clone(),
+        peer_store.clone(),
+        op_store,
+        mock_transport.clone(),
+    );
+
+    // Receive 2 op requests.
+    let requested_ops_1 = vec![op_id_1.clone(), op_id_2.clone()];
+    let requested_ops_2 = vec![op_id_3.clone(), op_id_4.clone()];
+    futures::future::join_all([
+        fetch.respond_with_ops(requested_ops_1, agent_id_1.clone()),
```

Comment:
I wonder if `request_op_data` might be a clearer name for this? I can't quite get this into my head

Reply:
I don't find the name very comprehensible either. But it's not a request, it's a response to a request which pulls ops from the store and sends them to the requesting agent.

Reply:
I see what you're saying but in my head it kind of is a request. We're choosing to queue the work in the implementation and this doesn't actually respond in the function return but it is still a remote peer requesting us to do work... Naming is hard :) 

Maybe something like `handle_op_data_request`?

Reply:
Yes, that's good and fits the `handle_incoming_ops` in the next PR. I'll change it to `handle_op_request`. I'd change `add_ops` to `request_ops` too.